fenix/app/src/main/java/org/mozilla/fenix/translations/TranslationSwitchItem.kt

```diff
@@ -8,14 +8,20 @@ package org.mozilla.fenix.translations
  * TranslationSwitchItem that will appear on Translation screens.
  *
  * @property textLabel The text that will appear on the switch item.
+ * @property description An optional description text below the label.
+ * @property importance Based on this, the translation switch item is enabled or disabled.
```

Comment:
nit:  From what I understand, `importance` only controls showing the `description`, is that right? Maybe adjust the language here since `importance` reads like it could impact the switch's `isEnabled` state too.

Reply:
Reread the PR: I do see some coupling on `importance` with other states. I think we might want to keep them decoupled, what do you think? 

Might also depend on  how we'd like the controller to send the initial state over.

Reply:
Based on importance, I enabled and disabled the views, or I showed the description.

Reply:
The logic here is complicated because of the design. It happens a lot of times, almost at random. The description is different for every view, and the enabled state is the same.

Reply:
Thanks for that background! Yeah, I agree the design is complicated here with a whole lot to consider. 

I wonder if we should consider if there is way to decouple the logic from the view and have a utility-like function that pre-calculates the state so that the view is purely consuming a data model?

Where it is so complicated, there is a lot to follow on how the view is making the determinations on if it should be switched or enabled. Also, it would be nice to be able to test this logic. 

We can also revisit how all of this works when connecting to the functionality too and have some of the controller logic setup.

I'm wondering maybe something like:

```
initialSettingState : PagePreferences = {
  globalAlways = true,
  languageAlways = true,
  languageNever = false,
  siteNever = false,
}
```


```
toggleSettings : TranslationSwitchItem = {
  textLabel,
  description,
  checked,
  enabled, 
  anything else we need for each view toggle to know
}
```

```
initialSettingState : PageSwitchSettings = {
  globalAlways : TranslationSwitchItem,
  languageAlways : TranslationSwitchItem,
  languageNever : TranslationSwitchItem,
  siteNever : TranslationSwitchItem,
}
```

```
computePageSwitchSettingsState(preferences: PagePreferences) : PageSwitchSettings {

  // Logic on how the initial settings determines toggle state in one place

  return PageSwitchSettings
}
```

Just thinking out loud! I think this might also be beneficial when we are connecting to the functionality too.

Reply:
We talked in Slack - we are going to follow-up on this when connecting to the functionality, so we have the full context.